src/EditorFeatures/CSharpTest/EditAndContinue/RudeEditStatementTests.cs

```diff
@@ -905,6 +905,529 @@ public void MatchExceptionHandlers()
+                { "a1", "a1" },
+                { "a2", "a3" },
+                { "() => { return 7; }", "() => { return 8; }" },
+                { "{ return 7; }", "{ return 8; }" },
+                { "return 7;", "return 8;" }
+            };
+
+            expected.AssertEqual(actual);
+        }
+
+        [Fact]
+        public void MatchParenthesizedVariable_Insert()
+        {
+            var src1 = @"var (z1, z2) = (1, 2);";
+            var src2 = @"var (z1, z2, z3) = (1, 2, 5);";
+
+            var match = GetMethodMatches(src1, src2, kind: MethodKind.Regular);
+            var actual = ToMatchingPairs(match);
+
+            var expected = new MatchingPairs
+            {
+                { "var (z1, z2) = (1, 2);", "var (z1, z2, z3) = (1, 2, 5);" },
+                { "z1", "z1" },
+                { "z2", "z2" }
+            };
+
+            expected.AssertEqual(actual);
+        }
+
+        [Fact]
+        public void MatchParenthesizedVariable_Delete()
+        {
+            var src1 = @"var (y1, y2, y3) = (1, 2, 7);";
+            var src2 = @"var (y1, y2) = (1, 4);";
+
+            var match = GetMethodMatches(src1, src2, kind: MethodKind.Regular);
+            var actual = ToMatchingPairs(match);
+
+            var expected = new MatchingPairs
+            {
+                { "var (y1, y2, y3) = (1, 2, 7);", "var (y1, y2) = (1, 4);" },
+                { "y1", "y1" },
+                { "y2", "y2" }
+            };
+
+            expected.AssertEqual(actual);
+        }
+
+        [Fact]
+        public void MatchForeachVariable_Update1()
+        {
+            var src1 = @"
+foreach (var (a1, a2) in e) { A1(); }
+foreach ((var b1, var b2) in e) { A2(); }
+";
+
+            var src2 = @"
+foreach (var (a1, a3) in e) { A1(); }
+foreach ((var b3, int b2) in e) { A2(); }
+";
+
+            var match = GetMethodMatches(src1, src2, kind: MethodKind.Regular);
+            var actual = ToMatchingPairs(match);
+
+            var expected = new MatchingPairs
+            {
+                { "foreach (var (a1, a2) in e) { A1(); }", "foreach (var (a1, a3) in e) { A1(); }" },
+                { "a1", "a1" },
+                { "a2", "a3" },
+                { "{ A1(); }", "{ A1(); }" },
+                { "A1();", "A1();" },
+                { "foreach ((var b1, var b2) in e) { A2(); }", "foreach ((var b3, int b2) in e) { A2(); }" },
+                { "b1", "b3" },
+                { "b2", "b2" },
+                { "{ A2(); }", "{ A2(); }" },
+                { "A2();", "A2();" },
+            };
+
+            expected.AssertEqual(actual);
+        }
+
+        [Fact]
+        public void MatchForeachVariable_Update2()
+        {
+            var src1 = @"
+foreach (_ in e2) { yield return 4; }
+foreach (_ in e3) { A(); }
+";
+
+            var src2 = @"
+foreach (var b in e2) { yield return 4; }
+foreach (_ in e4) { A(); }
```

Comment:
>in e4) { A(); } [](start = 11, length = 15)

It would be interesting here to move this foreach in front of the previous one. The match should not change.